src/accessor/grib_accessor_class_grid_spec.cc

```diff
@@ -30,8 +46,51 @@ long grib_accessor_grid_spec_t::get_native_type()
     return GRIB_TYPE_STRING;
 }
 
-int grib_accessor_grid_spec_t::unpack_string(char* v, size_t* len)
+int grib_accessor_grid_spec_t::unpack_string(char* val, size_t* len)
 {
-    // TODO(mapm): Please fill in
+#if defined(HAVE_GEOGRAPHY) && defined(HAVE_ECKIT_GEO)
+    ECCODES_ASSERT(0 < *len);
+    ECCODES_ASSERT(val != nullptr);
+
+    auto* h = grib_handle_of_accessor(this);
+    ECCODES_ASSERT(h != nullptr);
+
+    std::string spec_str;
+
+    try {
+        eccodes::geo::eckit_main_init();
+
+        std::unique_ptr<const eckit::geo::Spec> spec(new eccodes::geo::GribSpec(h));
+        std::unique_ptr<const eckit::geo::Grid> grid(eckit::geo::GridFactory::build(*spec));
+
+        spec_str = grid->spec_str();
+    }
+    catch (eckit::geo::Exception& e) {
+        grib_context_log(context_, GRIB_LOG_ERROR, "grib_accessor_grid_spec_t: geo::Exception thrown (%s)", e.what());
+        return GRIB_GEOCALCULUS_PROBLEM;
+    }
+    catch (std::exception& e) {
+        grib_context_log(context_, GRIB_LOG_ERROR, "grib_accessor_grid_spec_t: Exception thrown (%s)", e.what());
+        return GRIB_GEOCALCULUS_PROBLEM;
+    }
+
+    // guarantee null-termination
+    auto length = spec_str.length();
+    if (*len < length + 1) {
+        grib_context_log(context_, GRIB_LOG_ERROR,
+                         "%s: Buffer too small for %s. It is %zu bytes long (len=%zu)",
+                         class_name_, name_, length, *len);
+        *len = length;
```

Comment:
I'm nut sure I should be doing this assignment, given the buffer is of static size (also on line 90)